src/assistant.js

```diff
@@ -154,78 +156,89 @@ export async function openAssistantForm({
+			if ([TASK_STATUS_STRING.scheduled, TASK_STATUS_STRING.running].includes(task?.status)) {
+				getTask(task.id).then(response => {
+					const updatedTask = response.data?.ocs?.data?.task
 
-						const setProgress = (progress) => {
-							view.progress = progress
+					if (![TASK_STATUS_STRING.scheduled, TASK_STATUS_STRING.running].includes(updatedTask?.status)) {
+						view.selectedTaskTypeId = updatedTask.type
+						view.inputs = updatedTask.input
+						view.outputs = updatedTask.status === TASK_STATUS_STRING.successful ? updatedTask.output : null
+						view.selectedTaskId = updatedTask.id
+						lastTask = updatedTask
+						return
+					}
+
+					getNotifyReady(task.id).then(response => {
+						view.isNotifyEnabled = !!response.data?.ocs?.data?.id
+					}).catch(error => {
+						console.error('[assistant] get task notification status error', error)
+					})
+
+					view.loading = true
+					view.showSyncTaskRunning = true
+					view.progress = null
+					view.expectedRuntime = (updatedTask?.completionExpectedAt - updatedTask?.scheduledAt) || null
+
+					const setProgress = (progress) => {
+						view.progress = progress
+					}
+					pollTask(updatedTask.id, setProgress).then(finishedTask => {
+						console.debug('pollTask.then', finishedTask)
+						if (finishedTask.status === TASK_STATUS_STRING.successful) {
+							view.outputs = finishedTask?.output
+							view.selectedTaskId = finishedTask?.id
+						} else if (finishedTask.status === TASK_STATUS_STRING.failed) {
+							showError(t('assistant', 'Your task with ID {id} has failed', { id: finishedTask.id }))
+							console.error('[assistant] Task failed', finishedTask)
+							view.outputs = null
 						}
-						pollTask(updatedTask.id, setProgress).then(finishedTask => {
-							console.debug('pollTask.then', finishedTask)
-							if (finishedTask.status === TASK_STATUS_STRING.successful) {
-								view.outputs = finishedTask?.output
-								view.selectedTaskId = finishedTask?.id
-							} else if (finishedTask.status === TASK_STATUS_STRING.failed) {
-								showError(t('assistant', 'Your task with ID {id} has failed', { id: finishedTask.id }))
-								console.error('[assistant] Task failed', finishedTask)
-								view.outputs = null
-							}
-							// resolve(finishedTask)
-							view.loading = false
-							view.showSyncTaskRunning = false
-						}).catch(error => {
-							console.debug('[assistant] poll error', error)
-						})
+						// resolve(finishedTask)
+						view.loading = false
+						view.showSyncTaskRunning = false
+						emit('assistant:task:updated', finishedTask)
 					}).catch(error => {
-						console.error(error)
+						console.debug('[assistant] poll error', error)
 					})
-				}
+				}).catch(error => {
+					console.error(error)
+				})
 			}
 		})
 		view.$on('new-task', () => {
 			console.debug('[assistant] new task')
+			cancelTaskPolling()
+			view.showSyncTaskRunning = false
+			view.isNotifyEnabled = false
```

Comment:
This is needed in `openAssistantTask()` as well.

Reply:
Is it possible to create a notification for a task that is still running/scheduled? I thought that shouldn't be the case.